# This file is used for describing the corridor in this task.
# number of path
num_path: 2
# data of the task
task_data:
# task 1
  # start point
  - start_pt:
      pos: [0.0, 1.5, -2.8]
      vel: [0.0, 0.0, 0.0]
      acc: [0.0, 0.0, 0.0]
  # end point 
    end_pt:
      pos: [0.0, -1.5, -2.8]
      vel: [0.0, 0.0, 0.0]
      acc: [0.0, 0.0, 0.0]
  # task type: 0->move, 1->dock, 2->manipulation 
    task_type: 1
  # holding time for end-effector working 
    hold_time: 10.0
  # number of object ID.  if end_eff_type> 4, it means the target is a manipulation target.
    end_eff_type: 2
  # end-effector is open or close. true-> open, false -> close. Only valide for the manipulaton.
    end_eff_switch: false
  # offset for docking and manipulation ID 1->2.8 ID 2- >
    end_dock_offset: 0.050
  # pick up or change end-effector : false --- pick, true --- change
  # if end_eff_type == 0, docking_pick_change is useless
    docking_pick_change: false
  # corridor 
    num_polyhedron: 1
    corridor:
  # corridor 1
      - s_scale: 15.0
        upper_vel: [1.0,1.0,1.0]
        lower_vel: [-1.0, -1.0, -1.0]
        upper_acc: [1.0,1.0,1.0]
        lower_acc: [-1.0, -1.0, -1.0]
        A_pos: 
          row: 3
          col: 3
          data: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0,0.0, 0.0, 1.0]
        b_u_pos: [1.0, 2.0, -1.0]
        b_l_pos: [-1.0, -2.0, -3.0]


# task 2
  # start point
  - start_pt:
      pos: [0.0, -1.5, -2.8]
      vel: [0.0, 0.0, 0.0]
      acc: [0.0, 0.0, 0.0]
  # end point 
    end_pt:
      pos: [0.0, 1.5, -2.8]
      vel: [0.0, 0.0, 0.0]
      acc: [0.0, 0.0, 0.0]
  # task type: 0->move, 1->dock, 2->manipulation 
    task_type: 0
  # holding time for end-effector working 
    hold_time: 0.0
  # number of task.  Only valide for docking and manipulation. if end_eff_type> 4, it means the target is a manipulation target.
    end_eff_type: 0
  # end-effector is open or close. true-> open, false -> close. Only valide for the manipulaton.
    end_eff_switch: false
  # offset for docking  and manipulation
    end_dock_offset: 0.0
  # pick up or change end-effector : false --- pick, true --- change
  # if end_eff_type == 0, docking_pick_change is useless
    docking_pick_change: false
  # corridor 
    num_polyhedron: 1
    corridor:
  # corridor 1
      - s_scale: 15.0
        upper_vel: [1.0,1.0,1.0]
        lower_vel: [-1.0, -1.0, -1.0]
        upper_acc: [1.0,1.0,1.0]
        lower_acc: [-1.0, -1.0, -1.0]
        A_pos: 
          row: 3
          col: 3
          data: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0,0.0, 0.0, 1.0]
        b_u_pos: [1.0, 2.0, -1.0]
        b_l_pos: [-1.0, -2.0, -3.0]